src/Symfony/Component/HttpClient/UriTemplate/CallbackUriTemplateExpander.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\UriTemplate;
+
+/**
+ * This implementation allows you to easily forward the call to a vendor implementation of your choice.
+ */
+class CallbackUriTemplateExpander implements UriTemplateExpanderInterface
+{
+    /**
+     * @param \Closure(string $url, array $variables): string $callback
+     */
+    public function __construct(private readonly \Closure $callback)
+    {
+    }
+
+    public function expand(string $url, array $variables): string
+    {
+        return ($this->callback)($url, $variables);
+    }
+
+    /**
+     * @internal
+     */
+    public static function createFromPopularVendors(): static
+    {
+        if (class_exists(\League\Uri\UriTemplate::class)) {
```

Comment:
Those seem to be the 3 more popular vendor implementations (https://packagist.org/?query=uri%20template)